### PR TITLE
FOUR-18929 Add support to use micro service in Vault server to encrypt/decrypt data

### PR DESCRIPTION
**Pull request by julceslauhub**
https://processmaker.atlassian.net/browse/FOUR-18929

ci:next

### CODE REVIEW

File: ProcessMaker/EncryptedData/Local.php [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+{
+    const ENCRYPTION_METHOD = 'aes-256-cbc';
+
+    public static $iv = '';
+
+    /**
+     * Encrypt text.
+     *
+     * @param string $plainText
+     * @param string $iv
```
Comment (caleeli) [inline ProcessMaker/EncryptedData/Local.php:21]: Update the documentation

```suggestion
     * @param string|null $iv
```

File: ProcessMaker/EncryptedData/Vault.php [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    public function __construct()
+    {
+        // Get config
+        $this->host = config('app.encrypted_data.vault_host');
+        $this->token = config('app.encrypted_data.vault_token');
+        $this->transitKey = config('app.encrypted_data.vault_transit_key');
+    }
+
+    /**
+     * Encrypt text.
```
Comment (caleeli) [inline ProcessMaker/EncryptedData/Vault.php:31]: Please complete the documentation, explain when the $iv and $key can be optional, when not.

File: ProcessMaker/EncryptedData/Vault.php [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+        // Encrypt text
+        $response = $transitApi->encryptData($encryptDataRequest);
+        $cipherText = $response->getCiphertext();
+
+        return $cipherText;
+    }
+
+    /**
+     * Decrypt text.
```
Comment (caleeli) [inline ProcessMaker/EncryptedData/Vault.php:52]: Please complete the documentation, explain when the $iv and $key can be optional, when not.

File: ProcessMaker/EncryptedData/EncryptedDataInterface.php [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace ProcessMaker\EncryptedData;
+
+interface EncryptedDataInterface
+{
+    public function encryptText(string $plainText, string $iv = null, string $key = null): string;
```
Comment (caleeli) [inline ProcessMaker/EncryptedData/EncryptedDataInterface.php:7]: Please add the documentation of the methods, explain when the $iv and $key can be optional, and when not.

Comment (julceslauhub) [reply ProcessMaker/EncryptedData/EncryptedDataInterface.php:20]: done

**Review by caleeli — APPROVED** [review]

**Pull request merged**